CMakeLists.txt

```diff
@@ -315,6 +310,33 @@ option(USE_OPENSSL "Enable use of OpenSSL if available" ON)
 option(SEND_TELEMETRY_DEFAULT "The default value for whether to send telemetry" ON)
 option(REGRESS_CHECKS "PostgreSQL regress checks through installcheck" ON)
 
+# Option to enable assertions. Note that if we include headers from a
+# PostgreSQL build that has assertions enabled, we might inherit that
+# setting without explicitly enabling assertions via the ASSERTIONS
+# option defined here. Thus, this option is mostly useful to enable
+# assertions when the PostgreSQL we compile against has it disabled.
+option(ASSERTIONS "Compile with assertion checks (default OFF)" OFF)
+
+if (NOT EXISTS ${PG_INCLUDEDIR}/pg_config.h)
+  message(FATAL_ERROR "Could not find pg_config.h in ${PG_INCLUDEDIR}. "
+    "Make sure PG_PATH points to a valid PostgreSQL installation that includes development headers.")
+endif ()
+
+file(READ ${PG_INCLUDEDIR}/pg_config.h PG_CONFIG_H)
+string(REGEX MATCH "#define USE_ASSERT_CHECKING 1" PG_USE_ASSERT_CHECKING ${PG_CONFIG_H})
```

Comment:
Just a thought, that this might fail in case if there are any weird padding or the define being formatted using tabs instead of spaces, etc

Reply:
Since pg_config.h is auto-generated, I doubt that it will change to use different format anytime soon. In that case we'll notice I guess.